linchpin/api/__init__.py

```diff
@@ -92,21 +150,33 @@ def run_playbook(self, pinfile, targets='all', playbook='provision'):
                         '{0}/{1}/{2}'.format(self.ctx.workspace,
                                     self.ctx.cfgs['evars']['layouts_folder'],
                                     pf[target]["layout"]))
-
-
-
+                # set the current target data
+                self.current_target_data = pf[target]
+                self.current_target_data["extra_vars"] = self.ctx.evars
```

Comment:
After some conversation with @samvarankashyap, it's clear that this current_target_data dictionary needs to copy the ctx.evars value.